src/udev/udevadm-wait.c

```diff
@@ -48,22 +49,25 @@ static int check_device(const char *path) {
 
         assert(path);
 
+        if (arg_wait_until == WAIT_UNTIL_REMOVED) {
+                r = laccess(path, F_OK);
+                if (r == -ENOENT)
+                        return true;
+                if (r < 0)
+                        return r;
+                return false;
+        }
```

Comment:
hmm, I don't understand this change? why do we need to handle this beforehand explicitly, can you elaborate?

I mean, if I point sd_device_new_from_path() to a dangling symlink or on an absent device should not matter?

i.e. sd_device_new_from_path() does "stat()" on the passed path, and if that results in ENOENT (as a dangling symlink) would, then the function will return ENODEV, which the previous code was already checking for?